doc/reference/usermode/index.rst

```diff
@@ -121,13 +147,13 @@ High-level Policy Details
 Broadly speaking, we accomplish these thread-level memory protection goals
 through the following mechanisms:
 
-- Any user thread will only have access to its own stack memory by default.
-  Access to any other RAM will need to be done on the thread's behalf through
-  system calls, or specifically granted by a supervisor thread using the
-  :ref:`memory_domain` APIs. Newly created threads inherit the memory domain
-  configuration of the parent. Threads may communicate with each other
-  by having shared membership of the same memory domains, or via kernel objects
-  such as semaphores and pipes.
+- Any user thread will only have access to a subset of memory by default,
+  typically its stack and program text. Access to any other RAM will need to be
+  done on the thread's behalf through system calls, or specifically granted by
+  a supervisor thread using the :ref:`memory_domain` APIs. Newly created
+  threads inherit the memory domain configuration of the parent. Threads may
+  communicate with each other by having shared membership of the same memory
+  domains, or via kernel objects such as semaphores and pipes.
```

Comment:
Probably it's worth to mention the `main()` case as you put in the commit msg.
```
On any system a user thread can do very little without at least
some partitions mapped, such as the C library partition. It makes
sense for main() to start attached to some default domain.
```

Reply:
This is the direction I would like to go in, but I'd like to wait on this part until this is actually implemented in code. The current code allows for a thread to not be a member of a memory domain at all, which we can for now interpret as some default domain with only stack/text/rodata access.

When I overhaul the code, a user thread will always be a member of some domain. A user thread can switch domains, but never be without one. And there would be (either always or controlled by kconfig) a default user domain which has essential stuff mapped, like library globals. Supervisor thread would not have any memory domain membership all their stuff would be mapped in the upper half of memory whose mappings never change. (currently a supervisor thread can be in a domain but it has no effect on its access permissions). But we're not there yet. As more VM-related stuff takes shape and gets merged, there will be more PRs against doc/ to capture it.